### PR TITLE
Add text area component

### PR DESCRIPTION
**Pull request by jiejasonliu**
Two variants:
- readonly
- not... readonly

Philosophy: 
- consumer should subscribe to `onChange` and save the state on their own
- consumer needs to know what is in the textarea at anytime since there's no "submit/enter" logic

text-area

### CODE REVIEW

File: src/components/text-area/text-area.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+export const TextArea = ({
+  lines,
+  initialText,
+  label,
+  placeholder,
+  readonly,
+  resizable,
+  onChange,
+}: TextAreaProps) => {
+  const [value, setValue] = useState(initialText);
```
Comment (derricklee43) [inline src/components/text-area/text-area.tsx:24]: Could we move this out of the component and up a level? So have a prop `value` instead `initialText`. With the import and export dialog, we will need to be able to set the text repeatedly

File: src/components/text-area/text-area.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  return (
+    <div className="c-text-area-wrapper">
+      {label && <label className="c-text-area-label">{label}</label>}
+      <textarea
+        className={`c-text-area ${readonly ? 'readonly' : ''}`}
+        rows={lines}
+        placeholder={placeholder}
```
Comment (derricklee43) [inline src/components/text-area/text-area.tsx:32]: That's nice that nothing shows up when the placeholder is undefined. I'll going to simplify that in mine, haha

**Review by derricklee43 — CHANGES_REQUESTED** [review]

File: src/components/text-area/text-area.tsx (hunk 1 above)
Comment (jiejasonliu) [reply src/components/text-area/text-area.tsx:24]: Yeah, I didn't think about a way of doing that without using `useRef` but I think this makes sense now.

File: src/components/text-area/text-area.tsx (hunk 2 above)
Comment (jiejasonliu) [reply src/components/text-area/text-area.tsx:32]: Yup, as long as it's not a field that makes the component uncontrolled, `undefined` will act as if it just ignored the attribute.

File: src/components/text-area/text-area.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    return (
+      <div className="c-text-area-footer">
+        <LockIcon />
+        <label>readonly</label>
+      </div>
+    );
+  }
+
+  function onChangeHandler(e: ChangeEvent<HTMLTextAreaElement>) {
+    onChange?.(e.target.value);
```
Comment (derricklee43) [inline src/components/text-area/text-area.tsx:50]: I just noticed this. That's so dope that you can use the optional chaining operator with function calls. I'm still this too😂

File: src/components/text-area/text-area.tsx (hunk 2 above)
Comment (derricklee43) [reply src/components/text-area/text-area.tsx:30]: That makes it easier on our end, haha

**Review by derricklee43 — APPROVED** [review]
LGTM!

File: src/components/text-area/text-area.tsx (hunk 3 above)
Comment (jiejasonliu) [reply src/components/text-area/text-area.tsx:50]: ahahaha wow, you're stealing everything!!! 😡😂

**Pull request merged**